src/metax_api/services/file_service.py

```diff
@@ -855,10 +857,16 @@ def _get_directory_contents_for_catalog_record(cls, directory_id, cr_id, file_na
             # didnt exist, or it was not selected
             raise Http404
 
-        dirs = Directory.objects.filter(id__in=directory_ids, directory_name__icontains=directory_name) \
-            .values(*directory_fields)
-        files = None if dirs_only else File.objects.filter(id__in=file_ids, file_name__icontains=file_name) \
-            .values(*file_fields)
+        dirs = Directory.objects.filter(id__in=directory_ids).values(*directory_fields)
+
+        # icontains returns exception on None and with empty string does unnecessary db hits
+        if directory_name:
+            dirs = dirs.filter(directory_name__icontains=directory_name)
+
+        files = None if dirs_only else File.objects.filter(id__in=file_ids).values(*file_fields)
+
+        if file_name:
+            files = files.filter(file_name__icontains=file_name)
```

Comment:
Will this crash if ``dirs_only`` is ``True``, and ``file_name`` is defined, since the last line is then essentially ``None.filter(...)``? Or can this combination exist?

Reply:
That's a bug spotted and now corrected. Related test added.